setup.py

```diff
@@ -168,6 +168,7 @@ def get_exts_for(name):
             "typing; python_version<'3.5'",
             "protobuf>=3",
             "tenacity>=5",
+            "attrs>=20.3.0",
```

Comment:
do we require at least `20.3.0`?

We should set this to be the minimum version that is supported.

e.g. if we all the apis we use have been around since `attrs==15.0.0` then we can probably just do `attrs`, or `attrs>=<min-version>`

since we don't want to force all of our customers to be using `20.3.0` if they have a hard requirement to be using `19.3.0`, and `19.3.0` will work just fine for us.

Reply:
Good catch, I'll look at which is the min version we should support 😄

Reply:
@brettlangdon from what I've looked at, the tracing library only ever uses `attr.s` or `attr.ib` to decorate and define attributes within classes, which has been around since... the beginning lol. From our vendored `attr.__init__.py`, the defined API includes functions that have been around since at least 20.1.0 (can go as far back as 17.1.0 technically). 

Plus, the attrs library goes as far to as that "Put simply, you shouldn’t ever be afraid to upgrade attrs if you’re only using its public APIs." It seems like we can likely set any version as the minimum version to support since we only care about `attr.ib` and `attr.s` unless there's something I've missed. 😃

Reply:
if we only ever use `attr.s` or `attr.ib` and they have been around since `>=15`, then we can likely just set `attrs` as the `install_requires`, that way we either grab the most recent or use whatever version our customers need

Reply:
```suggestion
            "attrs",
```

Reply:
Good point - I'll set the `install_requires` to use any `attrs`, but also add testing for `attrs>=15.0.0` and `latest` in our riotfile just to be safe.

Reply:
Update: turns out that the keyword arguments in `attr.ib()` have been getting updated, meaning the earliest version we can support is 19.2.0 with the `eq` keyword arg being added in that release (currently used in `ddtrace/internal/service`).